File: sorting Algorithm/2. BubbleSort.ts
function BubbleSort(nums: number[]) {
  let n = nums.length;

  for (let i = n - 1; i >= 1; i--) {
    for (let j = 0; j <= i - 1; j++) {
      if (nums[j] > nums[j + 1]) {
        [nums[j], nums[j + 1]] = [nums[j + 1], nums[j]];
      }
    }
  }

  return nums;
}

console.log(BubbleSort([13, 46, 24, 52, 20, 9]));

File: sorting Algorithm/2. BubbleSort.py
def BubbleSort(nums):
    n = len(nums)
    for i in range(n-1, -1, -1):
        didSwap = 0
        for j in range(0, i):
            if nums[j] > nums[j+1]:
                [nums[j], nums[j+1]] = [nums[j+1], nums[j]]
                didSwap = 1

        if didSwap == 0:
            break
    return nums


# print(BubbleSort([13, 46, 24, 52, 20, 9, 10]))
print(BubbleSort([2, 5, 10, 15, 20]))
